app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts

```diff
@@ -41,6 +41,7 @@ class AppengineServerGroupBasicSettingsCtrl implements IController {
     return this.$scope.command.sourceType === AppengineSourceType.GCS;
   }
 
+  // TODO(jtk54): this is a copy of core code, please dedup using expected-artifact-selector component
```

Comment:
Might need a `service`.

Reply:
I think the directive will be enough, since it handles the rendering logic too